### PR TITLE
Feature/add attach scope example

### PR DESCRIPTION
**Pull request by gabrielgiroe1**
https://github.com/avo-hq/avo/issues/2648
https://github.com/avo-hq/avo/issues/2648#issuecomment-2045665943

### CODE REVIEW

File: docs/3.0/common/associations_attach_scope_option_common.md [hunk 1]
```diff
@@ -13,10 +13,23 @@ field :user,
...
 
 :::warning
 The `attach_scope` will not filter the records in the listing from `has_many` or `has_and_belongs_to_many` associations.
 Use [`scope`](#scope) or a [Pundit policy `Scope`](./../authorization#scopes) for that.
+
+Pass in a block where you attach scopes to the `query` object. The block is executed in the [`ExecutionContext`](./../execution-context).
+
 :::
+
+```ruby{3}
```
Comment (adrianthedev) [inline docs/3.0/common/associations_attach_scope_option_common.md:28]: You highlight the line number 3. Is there anything special about that line? Did you mean line number 5?

File: docs/3.0/common/associations_attach_scope_option_common.md [hunk 2]
```diff
@@ -13,10 +13,23 @@ field :user,
...
 
+
 Pass in a block where you attach scopes to the `query` object and `parent` object, which is the actual record where you want to assign the association. The block is executed in the [`ExecutionContext`](./../execution-context).
 :::
 
 :::warning
 The `attach_scope` will not filter the records in the listing from `has_many` or `has_and_belongs_to_many` associations.
 Use [`scope`](#scope) or a [Pundit policy `Scope`](./../authorization#scopes) for that.
+
+Pass in a block where you attach scopes to the `query` object. The block is executed in the [`ExecutionContext`](./../execution-context).
```
Comment (adrianthedev) [inline docs/3.0/common/associations_attach_scope_option_common.md:24]: I don't understand this sentence. I mean, what should the user take from it?

File: docs/3.0/common/associations_attach_scope_option_common.md [hunk 3]
```diff
@@ -13,10 +13,23 @@ field :user,
   attach_scope: -> { query.non_admins }
 ```
 
+
```
Comment (adrianthedev) [inline docs/3.0/common/associations_attach_scope_option_common.md:16]: Is this needed?
```suggestion
```

File: docs/3.0/common/associations_attach_scope_option_common.md [hunk 4]
```diff
@@ -13,10 +13,23 @@ field :user,
...
 :::
+
+```ruby{3}
+field :members,
+  as: :has_many,
+  through: :team_memberships,
+  searchable: true,
+  attach_scope: -> { query.where.not(team_id: parent.id) }
+  ```
+This example ensures that when attaching members to a team, only those who are not already members of that team will appear in the list of options.
```
Comment (adrianthedev) [inline docs/3.0/common/associations_attach_scope_option_common.md:35]:
```suggestion
In this example, in the `attach_scope`, we ensure that when attaching members to a team, only those who are not already members will appear in the list of options.
```

**Review by adrianthedev — CHANGES_REQUESTED** [review]
Left a coupld of suggestions and questions

File: docs/3.0/common/associations_attach_scope_option_common.md (hunk 2 above)
Comment (gabrielgiroe1) [reply docs/3.0/common/associations_attach_scope_option_common.md]: My mistake. It was something I copied from above and forgot to delete.

File: docs/3.0/common/associations_attach_scope_option_common.md (hunk 3 above)
Comment (gabrielgiroe1) [reply docs/3.0/common/associations_attach_scope_option_common.md]: No, I will delete it.

**Pull request merged**